combit.ListLabel.SchemaAwareJsonDataProvider/JsonSchemaTable/JsonSchemaOnlyTableRow.cs

```diff
@@ -33,31 +33,34 @@ internal static JsonTableColumn ColumnFromSchemaData(string columnName, JsonSche
                 return new JsonTableColumn(columnName, typeof(string), LlConstants.NullValue);
             }
 
-            if (data.Type == JsonObjectType.Boolean)
+            bool isNullable = !data.IsRequired;
+
+            if (data.Type.HasFlag(JsonObjectType.Boolean))
             {
-                return new JsonTableColumn(columnName, typeof(bool), null);
+                return new JsonTableColumn(columnName, isNullable ? typeof(bool?) : typeof(bool), null);
             }
 
             //https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.7.3.1
-            if (data.Type == JsonObjectType.String && (data.Format == "date-time" || data.Format == "date" || data.Format == "time"))
+            if (data.Type.HasFlag(JsonObjectType.String) && (data.Format == "date-time" || data.Format == "date" || data.Format == "time"))
             {
-                return new JsonTableColumn(columnName, typeof(DateTime), null);
+                return new JsonTableColumn(columnName, isNullable ? typeof(DateTime?) : typeof(DateTime), null);
```

Comment:
Actually, on the List & Label side all types are nullable by default. Thus, whether a type is passed in as nullable or not shouldn't change the behavior. Can you elaborate a bit on why this change is required and why it's not added for the string type below? For symmetrical reasons I would have expected it there as well 🙂.

Reply:
If L&L types are null by default, this is not required. String is a reference type, so it can be null and older C# versions (<8) doesn't support nullable reference types.

Reply:
Yup, they are definitely nullable - you can pass NULL for any of our types out of the box. In this case, you could consider to push a change here?

Reply:
Resolved by your commit [b00a99a](https://github.com/combit/NETDataProviders/commit/b00a99a1a43df98dbe1e3a77f13ed24ce24d37ab)